### PR TITLE
Refactor e2e utility functions to a single separate package

### PR DESCRIPTION
**Pull request by periklis**
This PR builds upon #412 to address moving all E2E utility functions to a single separate package under `test/utils`. In addition it  collects common functions in logical groups each in a separate file (e.g. `WaitForDeployment` in `test/utils/wait.go`). Furthermore it replaces all operator-sdk test-framework deprecations with their recommended counterparts (e.g. `test.NewTestCtx` with `test.NewContext`).

**Notes for the reviewer:** Please take a look only on the last commit. The first commits is a separate PR #412.

/cc @ewolinetz

### CODE REVIEW

File: pkg/elasticsearch/client.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ const (
 )
 
 type Client interface {
+	GetClusterName() string
```
Comment (jcantrill) [inline pkg/elasticsearch/client.go:37]: I recognize this builds upon the established pattern but does not golang conventions typically lead us to just "ClusterName()" and maybe we should consider refactoring the others

File: test/e2e-olm/elasticsearch_test.go [hunk 2]
```diff
@@ -1,318 +1,484 @@
...
+}
 
-	elasticsearch "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
-	framework "github.com/operator-framework/operator-sdk/pkg/test"
-	v1 "k8s.io/api/core/v1"
-)
+func singleNodeTest(t *testing.T) {
+	f := test.Global
+
+	ctx := test.NewContext(t)
```
Comment (jcantrill) [inline test/e2e-olm/elasticsearch_test.go:34]: If we are going to split theses tests why not take the opportunty to refactor common functionality into comment functions

File: test/utils/configmaps.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package utils
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ConfigMap(name, namespace string, labels, data map[string]string) *corev1.ConfigMap {
```
Comment (jcantrill) [inline test/utils/configmaps.go:8]: This should be "NewConfigMap"  Do we not already have a configmap factory function in `pkg`

File: test/utils/elasticsearch.go [hunk 4]
```diff
@@ -0,0 +1,207 @@
+package utils
```
Comment (jcantrill) [inline test/utils/elasticsearch.go:1]: Consider placing in an `elasticsearch` package so you could import a call functions like 'elasticsearch.CreateCR`

File: test/e2e-olm/elasticsearch_test.go (hunk 2 above)
Comment (periklis) [reply test/e2e-olm/elasticsearch_test.go:34]: The test context is part of each test case to ensure isolation and cleanup per test case.

File: test/utils/elasticsearch.go (hunk 4 above)
Comment (periklis) [reply test/utils/elasticsearch.go:1]: Unfortunately, we have many places with packages refering to elasticsearch. I would reserve it for production code. The e2e test util code is manageable enough in one utils package and `utils.CreateElasticsearchCR` is indicative enough.

File: test/e2e-olm/elasticsearch_test.go (hunk 2 above)
Comment (periklis) [reply test/e2e-olm/elasticsearch_test.go:34]: FWIW, I suggest to keep grouping common functionality in a separate PR that addresses refactoring e2e to gingko/gomega which address BeforeEach/AfterEach/etc. more intuitively.

**Pull request closed without merging**